### ISSUE TITLE
apache-fakegooglebot - stdout: ''

### ISSUE THREAD

**Issue by frollic** [fail2ban/fail2ban#1194]
I get these in the fail2ban.log

```
2015-09-23 01:37:30,065 fail2ban.action         [1661]: ERROR   /etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot 5.39.222.253 -- stdout: ''
2015-09-23 01:37:30,066 fail2ban.action         [1661]: ERROR   /etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot 5.39.222.253 -- stderr: ''
2015-09-23 01:37:30,066 fail2ban.action         [1661]: ERROR   /etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot 5.39.222.253 -- returned 1
2015-09-23 01:37:30,462 fail2ban.actions        [1661]: NOTICE  [apache-fakegooglebot] Ban 5.39.222.253
```
matching entry in access_log is
```
5.39.222.253 - - [23/Sep/2015:01:37:29 +0200] "GET / HTTP/1.0" 200 49 "-" "Mozilla/5.0 (compatible; Googlebot/2.1; +http://www.google.com/bot.html)"
```
fail2ban comes from FC22:
fail2ban-0.9.3-1.fc22.noarch
fail2ban-server-0.9.3-1.fc22.noarch
fail2ban-sendmail-0.9.3-1.fc22.noarch
fail2ban-firewalld-0.9.3-1.fc22.noarch

**Comment by sebres** [comment]
This script can return 1 (without stdout/stderr) if host resolved for the IP adress does not match `crawl-.*\.googlebot\.com` (see https://github.com/fail2ban/fail2ban/blob/master/config/filter.d/ignorecommands/apache-fakegooglebot#L29).
I think, it can be expected situation, but at the moment I don't know how the logging of this "error" can be prevented.

@yarikoptic do you know by chance, it was always that way, or approximately it is result of #1155 (resp. #1187)?

**Comment by starrychloe** [comment]
Still happening.

```
2016-09-02 21:26:30,774 fail2ban.actions        [1341]: NOTICE  [apache-fakegooglebot] 139.59.31.211 already banned
2016-09-02 21:26:30,879 fail2ban.action         [1341]: ERROR   /etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot 139.59.31.211 -- stdout: ''
2016-09-02 21:26:30,880 fail2ban.action         [1341]: ERROR   /etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot 139.59.31.211 -- stderr: ''
2016-09-02 21:26:30,880 fail2ban.action         [1341]: ERROR   /etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot 139.59.31.211 -- returned 1
2016-09-02 21:26:30,881 fail2ban.filter         [1341]: INFO    [apache-fakegooglebot] Found 139.59.31.211
2016-09-02 21:26:31,776 fail2ban.actions        [1341]: NOTICE  [apache-fakegooglebot] 139.59.31.211 already banned
```

**Comment by derekatkins** [comment]
I'm seeing this on my fresh CentOS7 install with fail2ban-0.9.5-3.el7.noarch.  The good news is that it does appear to actually ban the bogus attempts.  but the extraneous errors are annoying.

**Comment by AlexisWilke** [comment]
In my case I am getting an import error:

> ImportError: No module named fail2ban.server.filter

And the problem when such happens is that all IPs, Google or not, get banned. So obvious I can fix my problem by removing that jail for now. But I'm wondering whether you could at least detect whether the script executes sanely (opposed to just exiting with 0 [match] or 1 [not a match].)

    2016-11-18 14:06:22,063 fail2ban.action         [1590]: ERROR   /etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot 95.211.189.171 -- stdout: b''
    2016-11-18 14:06:22,063 fail2ban.action         [1590]: ERROR   /etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot 95.211.189.171 -- stderr: b'Traceback (most recent call last):\n  File "/etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot", line 32, in <module>\n    is_googlebot(process_args(sys.argv))\n  File "/etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot", line 17, in process_args\n    from fail2ban.server.filter import DNSUtils\nImportError: No module named fail2ban.server.filter\n'
    2016-11-18 14:06:22,065 fail2ban.action         [1590]: ERROR   /etc/fail2ban/filter.d/ignorecommands/apache-fakegooglebot 95.211.189.171 -- returned 1

**Comment by sebres** [comment]
@AlexisWilke Your issue is exact the same as #1506 (wrong python version in script, was fixed with #1508).

**Comment by sebres** [comment]
Fixed in 0.10th branch only (because of several codes incompatibilities to 0.9, e. g. code of executeCmd moved to utils module, etc.)
If someone needs it in 0.9, it may be cherry picked (+ conflicts resolving) to 0.9 branch...
